components/event-service/internal/events/subscribed/events.go

```diff
@@ -36,13 +43,62 @@ type eventsClient struct {
 	triggerLister kneventinglister.TriggerLister
 }
 
+// waitForInformersSyncOrDie blocks until all informer caches are synced, or panics after a timeout.
+func waitForInformersSyncOrDie(f externalversions.SharedInformerFactory) {
+	log.Println("waiting for informers caches sync...")
+	ctx, cancel := context.WithTimeout(context.Background(), informerSyncTimeout)
+	defer cancel()
+
+	err := hasSynced(ctx, f.WaitForCacheSync)
+	if err != nil {
+		log.Fatalf("Error waiting for caches sync: %s", err)
+	}
+}
+
+type waitForCacheSyncFunc func(stopCh <-chan struct{}) map[reflect.Type]bool
+
+// hasSynced blocks until the given informer sync waiting function completes. It returns an error if the passed context
+// gets canceled.
+func hasSynced(ctx context.Context, fn waitForCacheSyncFunc) error {
+	// synced gets closed as soon as fn returns
+	synced := make(chan struct{})
+
+	// closing stopWait forces fn to return, which happens whenever ctx
+	// gets canceled
+	stopWait := make(chan struct{})
+	defer close(stopWait)
+
+	// close the synced channel if the `WaitForCacheSync()` finished the execution cleanly
+	go func() {
+		informersCacheSync := fn(stopWait)
+		res := true
+		for _, sync := range informersCacheSync {
+			if !sync {
+				res = false
+			}
+		}
+		if res {
+			close(synced)
+		}
+	}()
+
+	// wait for closure of the goroutine or return an error if it timed out
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case <-synced:
+	}
+
+	return nil
+}
+
 //NewEventsClient function creates client for retrieving all active events
 func NewEventsClient(client versioned.Interface) EventsClient {
 	informerFactory := externalversions.NewSharedInformerFactory(client, 0)
```

Comment:
@radufa I think with passing the Default Resync Duration to the factory the resync may happen automatically.